### PR TITLE
Support YAML::Serializable

### PR DESCRIPTION
**Pull request by Blacksmoke16**
Surprise!  Is branched off of #253 so needs that first.

I'll need to edit the README again, probably use same examples as the JSON, but say it could also be YAML.

EDIT:  Since we didnt have this feature before i updated the labels.

### CODE REVIEW

**Comment by drujensen** [comment]
@Blacksmoke16 wow, very cool!

**Comment by Blacksmoke16** [comment]
An update on this, am going to skip adding docs for this for now.

Plan on dealing with #223 by making a directory to split-up our README into one topic per file and will write up docs for YAML then.

**Comment by robacarp** [comment]
@Blacksmoke16 looks like this will need some manual intervention to bring up to date with the json stuff.

**Comment by Blacksmoke16** [comment]
rip.  I'll take a look when i get home.

File: spec/granite_spec.cr [hunk 1]
```diff
@@ -67,7 +67,76 @@ require "./spec_helper"
...
-            collection.should eq(%([{"name":"todo 1","priority":1},{"name":"todo 2","priority":2},{"name":"todo 3","priority":3}]))
+            collection.should eq %([{"name":"todo 1","priority":1},{"name":"todo 2","priority":2},{"name":"todo 3","priority":3}])
+          end
+        end
+      end
+
+      describe "YAML" do
+        context ".from_yaml" do
+          it "can create an object from YAML" do
+            yaml_str = %({"name": "yaml::anyReview","upvotes": 2, "sentiment": 1.23, "interest": 4.56, "published": true})
```
Comment (robacarp) [inline spec/granite_spec.cr]: should this be more classical yaml instead of basically json?

**Comment by robacarp** [comment]
Thanks @Blacksmoke16

**Pull request merged**